apps/services/auth-api/src/app/modules/delegations/delegations.controller.spec.ts

```diff
@@ -0,0 +1,393 @@
+    let prTypeModel: typeof PersonalRepresentativeType
+
+    let einstaklingarApi: EinstaklingarApi
+
+    const userKennitala = getFakeNationalId()
+
+    const user = createCurrentUser({
+      nationalId: userKennitala,
+      scope: ['@identityserver.api/authentication'],
+    })
+
+    beforeAll(async () => {
+      app = await setupWithAuth({
+        user,
+      })
+      server = request(app.getHttpServer())
+
+      prTypeModel = app.get<typeof PersonalRepresentativeType>(
+        'PersonalRepresentativeTypeRepository',
+      )
+
+      await prTypeModel.create(personalRepresentativeType)
+
+      apiScopeModel = app.get<typeof ApiScope>('ApiScopeRepository')
+      prModel = app.get<typeof PersonalRepresentative>(
+        'PersonalRepresentativeRepository',
+      )
+      prRightsModel = app.get<typeof PersonalRepresentativeRight>(
+        'PersonalRepresentativeRightRepository',
+      )
+      prRightTypeModel = app.get<typeof PersonalRepresentativeRightType>(
+        'PersonalRepresentativeRightTypeRepository',
+      )
+      prScopePermission = app.get<typeof PersonalRepresentativeScopePermission>(
+        'PersonalRepresentativeScopePermissionRepository',
+      )
+      einstaklingarApi = app.get(EinstaklingarApi)
+    })
+
+    afterAll(async () => {
+      await app.cleanUp()
+    })
+
+    describe('and given we have 3 valid, 1 not yet active and 1 outdate right types', () => {
+      type rightsTypeStatus = 'valid' | 'unactivated' | 'outdated'
+      type rightsType = [code: string, status: rightsTypeStatus]
+      const rightsTypes: rightsType[] = [
+        ['valid1', 'valid'],
+        ['valid2', 'valid'],
+        ['unactivated', 'unactivated'],
+        ['outdated', 'outdated'],
+      ]
+
+      beforeAll(async () => {
+        await prRightTypeModel.bulkCreate(
+          rightsTypes.map(([code, status]) => {
+            switch (status) {
+              case 'valid':
+                return getPersonalRepresentativeRightType(code)
+              case 'unactivated':
+                return getPersonalRepresentativeRightType(
+                  code,
+                  faker.date.soon(7),
+                )
+              case 'outdated':
+                return getPersonalRepresentativeRightType(
+                  code,
+                  faker.date.recent(5),
+                  faker.date.recent(),
+                )
+            }
+          }),
+        )
+      })
+
+      afterAll(async () => {
+        await prRightTypeModel.destroy({
+          where: {},
+          cascade: true,
+          truncate: true,
+          force: true,
+        })
+      })
+
+      describe.each([
+        [1, 0, 0],
+        [2, 0, 0],
+        [0, 0, 0],
+        [0, 1, 0],
+        [0, 0, 1],
+        [0, 1, 1],
+        [1, 1, 1],
+      ])(
+        'and given user has %d active clients with valid rights, %d active clients with outdate rights and %d active clients with unactivated',
```

Comment:
Are we using "clients" in many places like this? Would like to choose another word that isn't confused with OAauth2 client. (was super confused 😆 )